implementation/src/main/java/io/smallrye/config/SecretKeys.java

```diff
@@ -0,0 +1,30 @@
+package io.smallrye.config;
+
+import java.io.Serializable;
+import java.util.function.Supplier;
+
+class SecretKeys implements Serializable {
+    private static final ThreadLocal<Boolean> LOCKED = ThreadLocal.withInitial(() -> Boolean.TRUE);
+
+    static boolean isLocked() {
+        return LOCKED.get();
+    }
+
+    static void accessSecrets(Runnable runnable) {
```

Comment:
This API should be public IMO.

```java
public static void doUnlocked(Runnable r) { ... }
public static <T> T doUnlocked(Supplier<T> s) { ... }
public static <T, R> R doUnlocked(Function<T, R> f) { ... }
```

And as mentioned in my other comment, the opposite:
```java
public static void doLocked(Runnable r) { ... }
public static <T> T doLocked(Supplier<T> s) { ... }
public static <T, R> R doLocked(Function<T, R> f) { ... }
```

Reply:
Ok. I was also delegating this to SmallRyeConfig. I think it is easier as an API perspective to be "found" since you see the method there, but we can have both.

Reply:
It all depends on how many things you want to put on the root API.  Either way works for me I guess.

Reply:
For now just `accessSecrets(Runnable)` and `accessSecret(Supplier)`. We can add the `Function` as well. These look to me to be the most common ones that you will need. If you need something more specialised, you use `SecretKeys`.